lib/lrama/states.rb

```diff
@@ -546,14 +545,106 @@ def compute_default_reduction
       end
     end
 
+    def compute_follow_kernel_items
+      set = nterm_transitions
+      relation = compute_goto_internal_relation
+      base_function = compute_goto_bitmaps
+      Digraph.new(set, relation, base_function).compute.each do |goto, follow_kernel_items|
+        state, nterm, next_state = goto
+        transition = state.nterm_transitions.find {|shift, _| shift.next_sym == nterm }
+        state.follow_kernel_items[transition] = state.kernels.map.with_index {|kernel, i|
+          [kernel, Bitmap.to_bool_array(follow_kernel_items, state.kernels.count)]
+        }.to_h
+      end
+    end
+
+    def compute_goto_internal_relation
+      nterm_transitions.map {|goto1|
+        related_gotos = nterm_transitions.select {|goto2| has_internal_relation?(goto1, goto2) }
+        [goto1, related_gotos]
+      }.to_h
+    end
+
+    def compute_goto_bitmaps
+      nterm_transitions.map {|state, sym, next_state|
+        bools = state.kernels.map.with_index {|kernel, i| i if kernel.next_sym == sym && kernel.symbols_after_transition.all?(&:nullable) }.compact
```

Comment:
Note: 
* `kernel.next_sym == sym` comes from a part of Definition 3.16-1 ($`\varrho[d] = \dagger(\sum[to\_stage[g']])`$)
* `kernel.symbols_after_transition.all?(&:nullable)` comes from Definition 3.16-2